### DISCUSSION TITLE
How to add a site thumbnail to the public library?

### DISCUSSION THREAD

**Discussion by maharvey** [fastaddons/GroupSpeedDial#308] [Q&A]
Less popular sites have no thumbnails in the public library, and sometimes the thumbnails are not of great quality, don't scale, or don't look good in dark mode. I typically hunt around to find a quality SVG image, but can only add it as an URL and it doesn't show up in the public library. Even if an image is not found in image search, I can often find the logo image embedded into a site's page and use that as an URL. In rare cases I've made my own image.

Is there any way to add an URL or file for a site thumbnail to the public library?

**Comment by Juraj-Masiar** [comment]
Hello,
I was actually thinking about this issue an hour ago when I was looking for a good svg image myself :).
In ideal world, the process of finding and adding image to the public library is as follows:
1. find the image
2. right click the image / "Use this image as dial thumbnail"
3. select the target dial
4. right click the dial / Advanced / "Upload thumbnail to public library"

There is potential catch though, in the step 2. Some images, when right clicked, are missing the options to use as thumbnail.
It's because the image is either set as background of some container, or the SVG image is directly integrated into the page (not loaded as a file).
In those cases, I usually look for the website on the wikipedia, there is often the official logo, I download it as SVG file, then I copy the file, open "Edit dial" window, press "Ctrl + v" which will set it as custom thumbnail. Then I continue with the step 4 :).

This is pretty terrible as you can see, so I do plan some upgrades...
I have already written a "logo extractor" that can find logos in the pages, but I haven't finished it and I need to integrate it to addon, so I've been postponing it.
I'll try to put it much higher on my todo list, since having a nice logos is always a good look, so I should make it easy to find one, even if the page is not popular abroad, or at all!

**Comment by HomeHeartTherapy** [comment]
Here is an idea for the situations wherein the logo isn't a separate file but instead embedded into the webpage.  Like in OneNote, just hit a button to take a "screenshot" that you can copy / paste directly into OneNote.  I use quotes because, traditionally, you would have to take a real screenshot, photo edit it (crop or even resize sometimes), maybe do some anti-aliasing with it, etc.  THEN you can upload it as a file.

But with OneNote, you just hit a button and the a "resize" box shows up (like a crop box) and then you just select exactly what you want.  Then hit another button and DONE!  Into OneNote it goes!  But in this use case, it would just be the dial container!

I think that should work, no?

New user here btw.  Just wanted to say thank you.  I'm still getting used to it but I love it.

My BIG ask is for someone to create a tool that can "scrape" my own 1000s of bookmarks (over the years), combine and organize them for me.  Maybe even have "linked" bookmarks.  For example, I might put a coding IDE YouTube video under "AI Tools\AI Coding IDEs" but then also want it under the "Programming\AI Coding\IDE bookmark directory.  Some bookmarks will go under several!  But wouldn't it be great to just bookmark it once?  And then an AI says...wait a minute!  You normally but anything with that subject into X, Y, and Z!  So I'll just do it once on a master list that you never look at but link them to each place as if you manually put them in those places yourself!

One can dream ;)

Thanks again!

**Reply by Juraj-Masiar** [reply]
The logos idea sounds nice, but this is crazy complicated thing to do.
But if you are using Windows 11, it's actually already "built-in" (more less).
Try this:
1. press PrintScreen button
2. this should open the "Snipping Tool" app, already in "select/crop" mode
3. mark the area you want to capture
4. at this point, the selected area should be already copied to clipboard (up until this point, this is all Windows 11 built-in feature)
5. right click the dial where you want to use this as image
6. select "Edit dial"
7. press "Ctrl + v" (there is a special code that detects image in the clipboard and inserts it)

Regarding the automatic organization - there is something like this on my todo list, but I don't think it's possible with my resources just yet.
